server/db/interface.go

```diff
@@ -0,0 +1,186 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package db
+
+import (
+	"github.com/decred/dcrdex/server/account"
+	"github.com/decred/dcrdex/server/market/types"
+	"github.com/decred/dcrdex/server/order"
+)
+
+type DEXArchivist interface {
+	OrderArchiver
+}
+
+type OrderArchiver interface {
+	// Order retrieves an order with the given OrderID, stored for the market
+	// specified by the given base and quote assets.
+	Order(oid order.OrderID, base, quote uint32) (order.Order, types.OrderStatus, error)
+	// StoreOrder stores an order with the provided status.
+	StoreOrder(ord order.Order, status types.OrderStatus) error
+
+	// OrderStatus gets the status, ID, and filled amount of the given order.
+	OrderStatus(order.Order) (types.OrderStatus, order.OrderType, int64, error)
+	// OrderStatusByID gets the status, ID, and filled amount of the order with
+	// the given OrderID in the market specified by a base and quote asset.
+	OrderStatusByID(oid order.OrderID, base, quote uint32) (types.OrderStatus, order.OrderType, int64, error)
+
+	// UpdateOrder updates the filled amount of the given order. This function
+	// applies only to market and limit orders, not cancel orders.
+	UpdateOrderFilled(order.Order) error
+	// UpdateOrderByID updates the filled amount of the order with the given
+	// OrderID in the market specified by a base and quote asset. This function
+	// applies only to market and limit orders, not cancel orders.
+	UpdateOrderFilledByID(oid order.OrderID, base, quote uint32, filled int64) error
+
+	// UpdateOrderStatus updates the status and filled amount of the given
+	// order.
+	UpdateOrderStatus(order.Order, types.OrderStatus) error
+	// UpdateOrderStatusByID updates the status and filled amount of the order
+	// with the given OrderID in the market specified by a base and quote asset.
+	// For cancel orders, the filled amount is ignored.
+	UpdateOrderStatusByID(oid order.OrderID, base, quote uint32, status types.OrderStatus, filled int64) error
+
+	// UserOrders retrieves all orders for the given account in the market
+	// specified by a base and quote asset.
+	UserOrders(aid account.AccountID, base, quote uint32) ([]order.Order, error)
+}
```

Comment:
The initial implementation of the `OrderArchiver interface` is `server/db/driver/pg.Archiver` at https://github.com/decred/dcrdex/pull/47/files#diff-884dd3a776297ea8952dcb641648bd62R62-R65